src/Http/Routing/src/Matching/CandidateSet.cs

```diff
@@ -281,6 +281,57 @@ public void SetValidity(int index, bool value)
             }
         }
 
+        /// <summary>
+        /// Replaces the <see cref="Endpoint"/> at the provided <paramref name="index"/> with the
+        /// provided <paramref name="endpoint"/>.
+        /// </summary>
+        /// <param name="index">The candidate index.</param>
+        /// <param name="endpoint">
+        /// The <see cref="Endpoint"/> to replace the original <see cref="Endpoint"/> at
+        /// the <paramref name="index"/>. If <paramref name="endpoint"/> the candidate will be marked
+        /// as invalid.
+        /// </param>
+        /// <param name="values">
+        /// The <see cref="RouteValueDictionary"/> to replace the original <see cref="RouteValueDictionary"/> at
+        /// the <paramref name="index"/>.
+        /// </param>
+        public void ReplaceEndpoint(int index, Endpoint endpoint, RouteValueDictionary values)
```

Comment:
I felt that it made most since for this to be a 1:1 replacement - there's no ability for a policy to inject 8 additional endpoints. We can add that in the future if someone really needs it.

Reply:
Would a non 1:1 replacement be useful for static files? I think that we should try and create a solution that can be used in the future to solve this issue - https://github.com/aspnet/AspNetCore/issues/4215

Background: I updated AuthSamples to use the AuthZ middleware. Being able to map static files in a location to endpoints (or a single dynamic endpoint), and then put AuthZ metadata on the endpoints would have made the sample really clean.

Reply:
> Would a non 1:1 replacement be useful for static files?

No I would imagine static files would be 1:1. So you register one *dynamic* endpoint - and then **run code** to figure out what endpoint you actually want to return - but in the case of static files it would always be one because static files would *unambiguously* determine what file matched.

The use case for multiple endpoints would be so that you can collaborate with another policy that needs to run after yours. Suppose you wanted to filter your results by HTTP method.